### ISSUE TITLE
Localization?

### ISSUE THREAD

**Issue by R3dL3ad3r** [atomicptr/dauntless-builder-reforged#69]
Not a real issue, but would it be possible to localize the builder in other languages?

**Comment by atomicptr** [comment]
Possible yes, but I'd imagine maintaining this to be a nightmare.

**Comment by R3dL3ad3r** [comment]
Hi there, sorry for the delay, i had some issues irl. I'm part of the localization team for the game and we would be interested in working on this builder too so the italian community can use it (sadly enough, most players find it difficult to use english material).
Please let us know if we can help you with an italian translation, we'd be happy to work on it.

**Comment by atomicptr** [comment]
Hey! This is def. something that I want to tackle with the "Dauntless Builder 2.0" rework, though I haven't had too much time recently myself.

There are generally a few approaches on how one could do this,

The most straightforward would probably having something like an overlay file that tries to load something in the requested language and should it fail, uses the english one instead, though that would mean either defining keywords like: INFERNOS_RAZOR_DESCRIPTION etc, or using the english one as a key "This weapon does.... blabla" (which sadly also has the disadvantage of breaking if the thing changes).

Such an overlay file could look like this:

```yaml
# de.translations.yml
translations:
   "Infernos Razor": Brennendes Schwert # not the actual name

# or in the more key based approach
translations:
  INFERNOS_RAZOR: Brennendes Schwert
```

Or defining everything in the original file as well like this:

```yaml
name:
  en: Infernos Razor
  de: Ein deutscher Name der mir nicht bekannt ist
  fr: etc etc
description:
  en: ...
````

Beyond me needing to find time, I'll obviously also have to find a format that a contributor (like yourself maybe) could work with.

**Comment by andres-conforti** [comment]
Would be great to get some other languages, not sure how many ppl are maintaining the website, but some help from the community would be great for that.

I'm in for the Spanish version if you need any help!

**Comment by R3dL3ad3r** [comment]
On a coding level i don't know what's the best way to do this, my C# knowledge isn't that great, but we're up for the translation side.

**Comment by atomicptr** [comment]
That was just me thinking out loud on possible ways to do this, it would be good to find a format with which people from the translation team could work with easily (a few more have contacted me via mail). I don't know how PhxLabs structures their data for translations, but it might be sensible to pick something thats similar.

**Comment by NaGaSiS** [comment]
Good afternoon, I offer to help with Spanish.
I think you should have a file for each language in json format that contains the necessary arrays for each language and all structures have a standard format.
That format equals the variable of the array that contains the language, so each file would be easy to modify by the assistants.

**Comment by R3dL3ad3r** [comment]
> That was just me thinking out loud on possible ways to do this, it would be good to find a format with which people from the translation team could work with easily (a few more have contacted me via mail). I don't know how PhxLabs structures their data for translations, but it might be sensible to pick something thats similar.

Phx Labs works with Crowdin, they drop strings almost daily for us translators to work with. I'm afraid i can't be more specific than that, as we only see the strings and a few context data, nothing more.

**Comment by atomicptr** [comment]
Resolved with #224
